app/controllers/application_controller.rb

```diff
@@ -59,6 +59,8 @@ def set_visitor_id
       visitor_id = SecureRandom.hex(26)
       cookies.permanent[:visitor_id] = { value: visitor_id, httponly: true }
     end
+    # If we run into cases where the intake does not have an associated visitor_id persisted onto it,
```

Comment:
Consider deleting the `visitor_record` method and calling `current_intake` here instead, since `visitor_record` is just a wrapper for `current_intake`.

If the record is persisted, then it'll always have a `visitor_id`, so consider removing this case.

If the record is not persisted yet, then it's good to tag the `current_intake` with the `visitor_id` from the cookie. I presume that's handled elsewhere.

None of the above is a blocker.

Reply:
yeah, i think that might clear some things up. visitor_record gets called from some other methods, but if its easy to transition into current_intake i will do that.

Reply:
unfortunately, I just checked on this and visitor_record is overridden with some other "intake types" in other controllers -- for example, current_diy_intake, which I guess is stored distinctly from current_intake. So I think for now, leaving visitor_record is a necessity.